### PR TITLE
Implemented a new metric called SegmentationPurityCoverageFMeasure.

### PR DESCRIPTION
**Pull request by diego-fustes**
The intention of this new metric is to serve as the same purpose as the
diarization counter-part, but in this case for segmentation jobs such as
Speaker Change Detection

### CODE REVIEW

File: pyannote/metrics/segmentation.py [hunk 1]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+        When provided, preprocess reference by filling intra-label gaps shorter
+        than `tolerance` (in seconds).
+
+    beta : float, optional
+            When beta > 1, greater importance is given to coverage.
+            When beta < 1, greater importance is given to purity.
+            Defaults to 1.
+
+    See also
+        --------
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
--------
```

File: pyannote/metrics/segmentation.py [hunk 2]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+        pyannote.metrics.segmentation.SegmentationPurity
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
pyannote.metrics.segmentation.SegmentationPurity
```

File: pyannote/metrics/segmentation.py [hunk 3]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+        pyannote.metrics.segmentation.SegmentationCoverage
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
pyannote.metrics.segmentation.SegmentationCoverage
```

File: pyannote/metrics/segmentation.py [hunk 4]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+        pyannote.metrics.base.f_measure
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
pyannote.metrics.base.f_measure
```

File: pyannote/metrics/segmentation.py [hunk 5]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+
+        detail = self.init_components()
+
+        # cooccurrence matrix coverage
+        K = reference * hypothesis
+        detail[CVG_TOTAL] = np.sum(K).item()
+        detail[CVG_INTER] = np.sum(np.max(K, axis=1)).item()
+
+        # cooccurrence matrix purity
+        L = hypothesis * reference
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]: I guess there is no need for this (L is simply the transpose version of K).

Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
```

File: pyannote/metrics/segmentation.py [hunk 6]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+        detail[PTY_TOTAL] = np.sum(L).item()
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
        detail[PTY_TOTAL] = detail[CVG_TOTAL]
```

File: pyannote/metrics/segmentation.py [hunk 7]
```diff
@@ -155,6 +162,79 @@ def compute_components(self, reference, hypothesis, **kwargs):
...
+        detail[PTY_INTER] = np.sum(np.max(L, axis=1)).item()
```
Comment (hbredin) [inline pyannote/metrics/segmentation.py]:
```suggestion
        detail[PTY_INTER] = np.sum(np.max(K, axis=0)).item()
```

**Comment by diego-fustes** [comment]
Yes, good point, we can reuse the same coocurrence matrix which is more efficient. All good from my side

**Comment by hbredin** [comment]
Thanks.

Do you plan to add other stuff to pyannote.metrics in the coming days?
If not, I will prepare a new release 1.8 with your latest addition.

**Comment by diego-fustes** [comment]
No, you can proceed with the release.

**Comment by hbredin** [comment]
I just [released](http://pyannote.github.io/pyannote-metrics/changelog.html#version-1-8-2018-09-03) pyannote.metrics 1.8.

**Pull request merged**